File: code_snippets/ts/src/concepts/invocations/delayed.ts
import * as restate from "@restatedev/restate-sdk";
import { greetCounter, greeter } from "./utils";

// <start_delayed_call>
async function myRestateHandler(ctx: restate.Context) {
  // !focus(1:4)
  // From a Restate handler
  ctx
    .serviceSendClient(greeter)
    .greet({ greeting: "Hi" }, restate.rpc.sendOpts({ delay: { seconds: 1 } }));
}
// <end_delayed_call>

File: code_snippets/python/src/concepts/invocations/delayed.py
from datetime import timedelta

from restate import Service, Context

from src.concepts.invocations.utils import greet

my_service = Service("MyService")


# <start_delayed_call>
@my_service.handler()
async def my_handler(ctx: Context, arg):
    # !focus
    ctx.service_send(greet, arg="Hi", send_delay=timedelta(seconds=1))
# <end_delayed_call>
